File: 백준/Silver/10845. 큐/큐.py
import sys
input = sys.stdin.readline

queue = []

N = int(input())
for i in range(N):
    str = input().split()
    if(len(str) == 2):
        commend, k = str[0], str[1]
    else :
        commend = str[0]
        
    if commend == "push":
        queue.append(k)
    elif commend == "pop":
        if len(queue) == 0:
            print(-1)
        else :
            print(queue.pop(0))
    elif commend == "size":
        print(len(queue))
    elif commend == "empty":
        if len(queue) == 0:
            print(1)
        else :
            print(0)
    elif commend == "front":
        if len(queue) == 0:
            print(-1)
        else :
            print(queue[0])
        
    elif commend == "back":
        if len(queue) == 0:
            print(-1)
        else :
            print(queue[len(queue)-1])
            
        
        

File: 백준/Silver/10845. 큐/큐.java
import java.io.*;
import java.util.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = null; 
        StringBuilder sb = new StringBuilder();
        ArrayDeque<Integer> queue = new ArrayDeque<>();

        // StringBuilder, switch 구문과 3항연산자를 활용해 코드 단순화
        // 각 case 별 기능 로직 구현
        int N = Integer.parseInt(br.readLine());
        for (int i = 0; i < N; i++) {
            st = new StringTokenizer(br.readLine());
            String command = st.nextToken();
            
            switch (command) {
                case "push":
                    int num = Integer.parseInt(st.nextToken());
                    queue.offer(num);
                    break;

                case "pop":
                    sb.append(queue.isEmpty() ? -1 : queue.poll()).append('\n');
                    break;

                case "size":
                    sb.append(queue.size()).append('\n');
                    break;

                case "empty":
                    sb.append(queue.isEmpty() ? 1 : 0).append('\n');
                    break;

                case "front":
                    sb.append(queue.isEmpty() ? -1 : queue.peekFirst()).append('\n');
                    break;

                case "back":
                    sb.append(queue.isEmpty() ? -1 : queue.peekLast()).append('\n');
                    break;
            }
        }
        System.out.print(sb);
    }
}